src/main/scala/com/databricks/labs/overwatch/validation/DataReconciliation.scala

```diff
@@ -0,0 +1,236 @@
+    val validationPath = {
+      if (!path.startsWith("dbfs:") && !path.startsWith("s3") && !path.startsWith("abfss") && !path.startsWith("gs")) {
+         s"""dbfs:${path}"""
+      }else{
+        path
+      }
+    }
+
+    val pipelineSnapTime =  Pipeline.createTimeDetail(LocalDateTime.now(Pipeline.systemZoneId).toInstant(Pipeline.systemZoneOffset).toEpochMilli)
+    reconStatusArray.toSeq.toDS().toDF()
+      .withColumn("deployment_id", lit(reconRunId))
+      .withColumn("snapTS", lit(pipelineSnapTime.asTSString))
+      .withColumn("timestamp", lit(pipelineSnapTime.asUnixTimeMilli))
+      .write.format("delta")
+      .option("mergeSchema", "true")
+      .mode("append")
+      .save(s"""${validationPath}/report/${reportName}""")
+    println("Validation report has been saved to " + s"""${validationPath}/report/${reportName}""")
+  }
+
+
+
+private def getQuery(tableName: String,orgId: String,target:PipelineTable): String = {
+  val _keys = target.keys.filterNot(_ == "Overwatch_RunID")
+  val reconColymns = target.reconColumns
+  val finalColumns = if(!reconColymns.isEmpty) {
+    _keys++reconColymns
+  }else{
+    _keys
+  }
+  s"""select ${finalColumns.mkString(",")} from $tableName where organization_id = ${orgId} """
+
+}
+
+  private def getTableDF(query: String):DataFrame = {
+    try{
+      val filterDF = spark.sql(query)
+      filterDF
+    }catch {
+      case exception: Exception =>
+        println(s"""Exception: Unable to run the query ${query}"""+exception.getMessage)
+        spark.emptyDataFrame
+    }
+
+  }
+
+  private[overwatch] def getAllTargets(workspace: Workspace): ParArray[PipelineTable] = {
+    val b = Bronze(workspace)
+    val s = Silver(workspace)
+    val g = Gold(workspace)
+    (b.getAllTargets ++ s.getAllTargets ++ g.getAllTargets).filter(_.exists(dataValidation = true, catalogValidation = false)).par
+  }
+
+  private[overwatch] def performBasicValidation(sourceOrgIDArr:Array[String],targetOrgIDArr:Array[String]):Unit = {
+    println("Number of workspace in Source:" + sourceOrgIDArr.size)
+    println("Number of workspace in Target:" + targetOrgIDArr.size)
+    if(sourceOrgIDArr.size<1 || targetOrgIDArr.size<1){
+      val msg ="Number of workspace in source/target etl is 0 , Exiting"
+      println(msg)
+      throw new Exception(msg)
+    }
+
+  }
+
+  private[overwatch] def getConfig(sourceEtl: String, orgID: String): Workspace = {
+    Helpers.getWorkspaceByDatabase(sourceEtl, Some(orgID))
+  }
+
+
+  private[overwatch] def getAllOrgID(etlDB: String): Array[String] = {
+    try{
+      spark.table(s"${etlDB}.pipeline_report").select("organization_id").distinct().collect().map(row => row.getString(0))
+    }catch {
+      case e:Throwable=>
+        val msg = "Got exception while reading from pipeline_report ,"
+        println(msg+e.getMessage)
+        throw e
+    }
+
+  }
+
+
+
+}
```

Comment:
@sriram251-code  Please resolve the merge conflict

Reply:
done